### PR TITLE
Update the timeout_notifications scheduled tasks.

### PR DESCRIPTION
**Pull request by servingUpAces**
We found that if the notifications were in created or pending they are not purged from notifications.
- New bulk update method to set all notificaitons with:
  -  a status = created|sending|pending to temporary-failure
  - and is older then today minus SENDING_NOTIFICATIONS_TIMEOUT_PERIOD (in seconds)
- the scheduled task to timeout notifications use the new bulk update query.
- the task will be more efficient

### CODE REVIEW

File: app/dao/notifications_dao.py [hunk 1]
```diff
@@ -286,3 +290,18 @@ def dao_delete_notifications_and_history_by_id(notification_id):
...
     ).delete(synchronize_session='fetch')
+
+
+def dao_timeout_notifications(timeout_period_in_seconds):
+    # update all notifications that are older that the timeout_period_in_seconds
+    #  with a status of created|sending|pending
+    updated = db.session.query(Notification). \
+        filter(Notification.created_at < (datetime.utcnow() - timedelta(seconds=timeout_period_in_seconds))). \
+        filter(Notification.status.in_([NOTIFICATION_CREATED, NOTIFICATION_SENDING, NOTIFICATION_PENDING])). \
+        update({'status': NOTIFICATION_TEMPORARY_FAILURE}, synchronize_session=False)
```
Comment (leohemsted) [inline app/dao/notifications_dao.py]: need to update `updated_at` as well (and for NotificationHistory below)

Comment (servingUpAces) [inline app/dao/notifications_dao.py]: Nice catch

File: app/celery/scheduled_tasks.py [hunk 2]
```diff
@@ -111,21 +111,13 @@ def delete_invitations():
...
+    try:
+        updated = dao_timeout_notifications(current_app.config.get('SENDING_NOTIFICATIONS_TIMEOUT_PERIOD'))
+        if updated:
+            current_app.logger.info(
+                "Timeout period reached for {} notifications, status has been updated.".format(updated))
+    except Exception as e:
+        current_app.logger.exception(e)
+        current_app.logger.error(
+            "Exception raised trying to timeout notification skipping notification update."
+        )
```
Comment (leohemsted) [inline app/celery/scheduled_tasks.py]: dumb q - do we gain anything from this exception? before it was making sure it tried every notification but celery will still log task as failed if it just bubbles up - i'd just kill this try except block

**Comment by leohemsted** [comment]
if you kill that try catch, then 👍

**Pull request merged**